test/cmds/watch/watch.test.js

```diff
@@ -0,0 +1,148 @@
+const EventEmitter = require('events');
+const chokidar = require('chokidar');
+const watch = require('../../../dist/src/cmds/watch');
+const build = require('../../../dist/src/cmds/build/bundle');
+const fsUtils = require('../../../dist/src/utils/validate-fs');
+const miscUtils = require('../../../dist/src/utils/misc');
+
+describe('watch', () => {
+  describe('Watch a directory and its subdirectories for changes, and build when files are added or changed.', () => {
+
+    let watcherEmitter;
+
+    const mockArgv = {
+      src: 'index.js',
+      dist: 'dist',
+      outfileName: 'bundle.js',
+    };
+
+    beforeEach(() => {
+      jest.spyOn(chokidar, 'watch').mockImplementation(() => {
+        watcherEmitter = new EventEmitter();
+        watcherEmitter.add = () => undefined;
+        jest.spyOn(watcherEmitter, 'on');
+        jest.spyOn(watcherEmitter, 'add');
+        return watcherEmitter;
+      });
+    });
+
+    afterEach(() => {
+      jest.clearAllMocks();
+      jest.restoreAllMocks();
+      watcherEmitter = undefined;
+    });
+
+    it('successfully processes arguments from yargs', async () => {
+      const chokidarMock = jest.spyOn(chokidar, 'watch').mockImplementation(() => {
+        watcherEmitter = new EventEmitter();
+        watcherEmitter.add = () => undefined;
+        jest.spyOn(watcherEmitter, 'on');
+        jest.spyOn(watcherEmitter, 'add');
+        return watcherEmitter;
+      });
+      jest.spyOn(console, 'log').mockImplementation();
+      const validateDirPathMock = jest.spyOn(fsUtils, 'validateDirPath').mockImplementation(() => true);
+      const validateFilePathMock = jest.spyOn(fsUtils, 'validateFilePath').mockImplementation(() => true);
+      const validateOutfileNameMock = jest.spyOn(fsUtils, 'validateOutfileName').mockImplementation(() => true);
+      jest.spyOn(fsUtils, 'getOutfilePath').mockImplementation(() => 'dist/bundle.js');
+      const root = (
+        mockArgv.src.indexOf('/') === -1 ? '.' : mockArgv.src.substring(0, mockArgv.lastIndexOf('/') + 1)
+      );
+      await watch.handler(mockArgv);
+      expect(validateDirPathMock).toHaveBeenCalledWith(mockArgv.dist, true);
+      expect(validateFilePathMock).toHaveBeenCalledWith(mockArgv.src);
+      expect(validateOutfileNameMock).toHaveBeenCalledWith(mockArgv.outfileName);
+      expect(chokidarMock.mock.calls[0][0]).toBe(root);
```

Comment:
We know the value that `root` will have, and so we should skip the variable entirely and just pass `'.'` as a literal to the `expect(chokidarMock...)` call:
```suggestion
      expect(chokidarMock.mock.calls[0][0]).toBe('.');
```